libopae/src/event.c

```diff
@@ -618,14 +618,17 @@ fpga_result __FPGA_API__ fpgaRegisterEvent(fpga_handle handle,
 	struct _fpga_token *_token;
 	int err;
 
-	result = handle_check_and_lock(_handle);
-	if (result)
-		return result;
-
+	// lock event handle first as other event API functions only lock the
```

Comment:
We should follow the same ordering as other places which lock both a handle and an event_handle (handle first, followed by the event_handle).